js/views/navigation-accounts.js

```diff
@@ -31,22 +31,14 @@ define(function(require) {
 			if (!_.isUndefined(activeFolder)) {
 				return activeFolder;
 			}
-
-			// bad hack to navigate down the tree ...
 			var delimiter = activeAccount.get('delimiter');
-			folderId = atob(folderId);
-			activeFolder = activeAccount;
-			var parts = folderId.split(delimiter);
-			var k = '';
-			_.each(parts, function(p) {
-				if (k.length > 0) {
-					k += delimiter;
-				}
-				k += p;
-				var folders = activeFolder.folders || activeFolder.get('folders');
-				activeFolder = folders.filter(function(f) {
-					return f.id === btoa(k);
-				}).shift();
+
+			var first_part = atob(folderId).split(delimiter)[0];
+			activeFolder = activeAccount.get('folders').get(btoa(first_part));
+			activeFolder.attributes.folders.forEach(function(folder) {
+				if(folder.id == folderId) {
```

Comment:
actually there should also be a space after the if. Simply amend your last commit.